prepare for release (#111)
rustacean-station/rustacean-station.org: _episodes/046-alice-ryhl.md

---
title: "Tokio Ecosystem with Alice Ryhl"

> date: 2021-11-12T02:11:00Z

Reviewer: ```suggestion
date: 2021-11-12T18:25:00Z
```

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Timestamps 
- [@00:40] - Alice's Bio
- [@01:08] - Managing pull requests on GitHub
- [@05:00] - Alice's involvement in Tokio
- [@08:23] - Tokio's topics page
- [@11:06] - Alice's favorite part of contributing
- [@12:55] - Changes in Tokio since Alice joined
- [@16:52] - Measuring metrics
- [@19:38] - Cooperative & preemptive scheduling
- [@24:30] - [Diesel](https://diesel.rs/)
- [@25:45] - Definition of [blocking]((https://ryhl.io/blog/async-what-is-blocking/)
- [@27:37] - I/O threads
- [@31:21] - What are sleeping threads?
- [@33:41] - [Tokio Console](https://tokio.rs/blog/2021-09-console-dev-diary-1)
- [@41:14] - Pros and cons of using actors
- [@47:05] - Alice's academic background
- [@49:22] - Tokio's upcoming roadmap
- [@57:33] - Replacing epoll with io_uring

> - [@58:56] - [Axum](https://github.com/tokio-rs/axum), [Tower](https://github.com/tower-rs/tokio-tower), and [Loom](https://github.com/tokio-rs/loom)

Reviewer: ```suggestion
- [@58:56] - [Axum](https://github.com/tokio-rs/axum), [Tower](https://github.com/tower-rs/tower), and [Loom](https://github.com/tokio-rs/loom)
```
tokio-tower is a different thing :)

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Timestamps 
- [@00:40] - Alice's Bio
- [@01:08] - Managing pull requests on GitHub
- [@05:00] - Alice's involvement in Tokio
- [@08:23] - Tokio's topics page
- [@11:06] - Alice's favorite part of contributing
- [@12:55] - Changes in Tokio since Alice joined
- [@16:52] - Measuring metrics
- [@19:38] - Cooperative & preemptive scheduling
- [@24:30] - [Diesel](https://diesel.rs/)
- [@25:45] - Definition of [blocking]((https://ryhl.io/blog/async-what-is-blocking/)
- [@27:37] - I/O threads
- [@31:21] - What are sleeping threads?
- [@33:41] - [Tokio Console](https://tokio.rs/blog/2021-09-console-dev-diary-1)

> - [@39:07] - Constructs, actors, and actors within Tokio 

Reviewer: This says "actors" twice?
Also, what does "constructs" mean here?

- [@41:14] - Pros and cons of using actors
- [@47:05] - Alice's academic background
- [@49:22] - Tokio's upcoming roadmap
- [@57:33] - Replacing epoll with io_uring

> - [@58:56] - Axum, Tower & Loom

Reviewer: ```suggestion
- [@58:56] - Axum, Tower, and Loom
```

Reviewer: Also, can we link to each of these?

---
title: "Tokio Ecosystem with Alice Ryhl"
date: 2021-11-12T02:11:00Z
file: https://audio.rustacean-station.org/file/rustacean-station/rustacean-station-e046-alice-ryhl.mp3
duration: "1:02:51"
length: "61148368"
#reddit: (leave blank on initial publish, amend with link and uncomment this line after Reddit thread has been posted)
---

> Allen Wyma talks with [Alice Ryhl](https://ryhl.io/), one of the primary maintainer of the open source project [Tokio](https://tokio.rs/), a widely used Rust library.

Reviewer: ```suggestion
Allen Wyma talks with [Alice Ryhl](https://ryhl.io/), one of the maintainers of the open source project [Tokio](https://tokio.rs/).
```
I think Tokio is well-enough known that we don't need to describe it.

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Timestamps 
- [@00:40] - Alice's Bio

> - [@01:08] - Pull request's process in GitHub

Reviewer: ```suggestion
- [@01:08] - Managing pull requests on GitHub
```

- [@05:00] - Alice's involvement in Tokio
- [@08:23] - Tokio's topics page

> - [@11:06] - Alice's favorite part in the contribution

Reviewer: ```suggestion
- [@11:06] - Alice's favorite part of contributing
```

> - [@12:55] - Changes in Tokio's since Alice joined

Reviewer: ```suggestion
- [@12:55] - Changes in Tokio since Alice joined
```

- [@16:52] - Measuring metrics
- [@19:38] - Cooperative scheduling

> - [@22:52] - Cooperative & Preemptive scheduler 

Reviewer: I feel like these can probably be merged. Also, Preemptive shouldn't be capitalized.

- [@24:30] - [Diesel](https://diesel.rs/)

> - [@25:45] - Definition of [Blocking]((https://ryhl.io/blog/async-what-is-blocking/)

Reviewer: ```suggestion
- [@25:45] - Definition of [blocking]((https://ryhl.io/blog/async-what-is-blocking/)
```

- [@27:37] - I/O threads

> - [@31:21] - Meaning of sleeping threads

Reviewer: ```suggestion
- [@31:21] - What are sleeping threads?
```

- [@33:41] - [Tokio Console](https://tokio.rs/blog/2021-09-console-dev-diary-1)

> - [@39:07] - Constructs, Actors, Actics within Tokio 

Reviewer: I have a hard time following this. Also, do you mean actix?

> - [@41:14] - Pros and Cons of Actor patterns

Reviewer: ```suggestion
- [@41:14] - Pros and cons of using actors
```

- [@47:05] - Alice's academic background

> - [@49:22] - [Tokio upcoming roadmap](https://tokio.rs/blog/2020-12-tokio-1-0)

Reviewer: The link here is weird to me, since it's describing 1.0, which has already launched, so it's not really a roadmap?

> - [@57:33] - Replacing epoll with io_uring

Reviewer: Maybe link to the announcement post of tokio_io_uring?

Reviewer: Specifically, this blog post: https://tokio.rs/blog/2021-07-tokio-uring

> - [@58:56] - New web server - Axum, Tower or Loom

Reviewer: I don't quite follow this. Axum is for writing web services, but tower is not, and loom is definitely not. Is this more about "tell me about these three crates"?

- [@01:01:45] - Web frameworks for Rust 

> - [@01:05:57] - How to contact Alice?

Reviewer: ```suggestion
- [@01:05:57] - How to contact Alice
```
It's not a question ;)

## Timestamps 
- [@00:40] - Alice's Bio
- [@01:08] - Managing pull requests on GitHub
- [@05:00] - Alice's involvement in Tokio
- [@08:23] - Tokio's topics page
- [@11:06] - Alice's favorite part of contributing
- [@12:55] - Changes in Tokio since Alice joined
- [@16:52] - Measuring metrics
- [@19:38] - Cooperative & preemptive scheduling
- [@24:30] - [Diesel](https://diesel.rs/)
- [@25:45] - Definition of [blocking]((https://ryhl.io/blog/async-what-is-blocking/)
- [@27:37] - I/O threads
- [@31:21] - What are sleeping threads?
- [@33:41] - [Tokio Console](https://tokio.rs/blog/2021-09-console-dev-diary-1)

> - [@39:07] - Constructs, actors, and actors within Tokio 

Reviewer: This says "actors" twice?
Also, what does "constructs" mean here?
